doc/dynamic-environments/configuration.mkd

```diff
@@ -454,3 +477,46 @@ This will create the following directory structure:
 |-- app1_production  # app1 data repository, production branch
 |-- app1_develop     # app1 data repository, develop branch
 ```
+
+### Filtering branches
+
+#### Filtering branches - Exclude branches with a prefix
+
+If you would like to filter out all branches starting with and including "dev-".
+
+```yaml
+---
+sources:
+  app1_data:
+    remote: 'git://git-server.site/my-org/app1-hieradata'
+    basedir: '/etc/puppet/hieradata'
+    branch_filter: '^(?!dev-).*$'
+```
+
+#### Filter branches - Include branches with a prefix
+
+If you would like to only deploy branches starting with a prefix "prod-" followed by word characters. Note: this does NOT match the "prod-" branch exactly due to the \w+.
+
+```yaml
+---
+sources:
+  app1_data:
+    remote: 'git://git-server.site/my-org/app1-hieradata'
+    basedir: '/etc/puppet/hieradata'
+    branch_filter: '^(prod-)\w+$'
+```
+
+#### Filter branches - Match branches exactly
+
+If you would like to only deploy branches whose name matches exactly to 'master', or 'production'.
+
+```yaml
+---
+sources:
+  app1_data:
+    remote: 'git://git-server.site/my-org/app1-hieradata'
+    basedir: '/etc/puppet/hieradata'
+    branch_filter: '^(master)|(production)$'
```

Comment:
This can be `'^(master|production)$'`.

Reply:
Yes, but it also doesn't matter. We're not actually using the group information so having 1 or 2 groups makes no difference.